rust-runtime/aws-smithy-runtime-api/src/http/headers.rs

```diff
@@ -181,12 +181,11 @@ impl TryFrom<http_02x::HeaderMap> for Headers {
     type Error = HttpError;
 
     fn try_from(value: http_02x::HeaderMap) -> Result<Self, Self::Error> {
-        if let Some(e) = value
+        if let Some(not_utf8) = value
             .values()
-            .filter_map(|value| std::str::from_utf8(value.as_bytes()).err())
-            .next()
+            .find(|value| std::str::from_utf8(value.as_bytes()).is_err())
         {
-            Err(HttpError::header_was_not_a_string(e))
+            Err(HttpError::not_utf8(not_utf8.as_bytes()))
```

Comment:
seems like including the actual header here (including the key!) would be more helpful

Reply:
We can't always include the key because some header-related methods focus solely on the value. For cases where the key _is_ available, I've added it to the error message.